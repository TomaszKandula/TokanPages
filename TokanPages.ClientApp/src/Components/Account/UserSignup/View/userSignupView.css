.user-signup-view-skeleton {
    width: 33vw;
}

@media (max-width: 768px) {
    .user-signup-view-column {
        margin-bottom: 3rem;
    }
}

@media (max-width: 768px) {
    .user-signup-view-card {
        width: 100vw;
    }
}
