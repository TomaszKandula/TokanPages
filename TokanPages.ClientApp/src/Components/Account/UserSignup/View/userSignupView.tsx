import * as React from "react";
import { Link } from "react-router-dom";
import Container from "@material-ui/core/Container";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import { Card, CardContent, CircularProgress } from "@material-ui/core";
import TextField from "@material-ui/core/TextField";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import { AccountCircle } from "@material-ui/icons";
import Skeleton from "@material-ui/lab/Skeleton";
import { VioletCheckbox } from "../../../../Theme";
import { UserSignupStyle } from "./userSignupStyle";

interface IBinding 
{
    bind: IProperties;
}

interface IProperties
{
    isLoading: boolean;
    caption: string;
    consent: string;
    button: string;
    link: string;
    buttonHandler: any;
    keyHandler: any;
    formHandler: any;
    progress: boolean;
    firstName: string;
    lastName: string;
    email: string;
    password: string;
    terms?: boolean;
    labelFirstName: string;
    labelLastName: string;
    labelEmail: string;
    labelPassword: string;
}

const ActiveButton = (props: IBinding): JSX.Element => 
{
    const classes = UserSignupStyle();
    return(
        <Button 
            fullWidth 
            type="submit" 
            variant="contained" 
            onClick={props.bind?.buttonHandler} 
            className={classes.button} 
            disabled={props.bind?.progress}>
            {!props.bind?.progress 
            ? props.bind?.button 
            : <CircularProgress size={20} />}
        </Button>
    );
}

const RedirectTo = (args: { path: string, name: string }): JSX.Element => 
{
    return(<Link to={args.path}>{args.name}</Link>);
}

export const UserSignupView = (props: IBinding): JSX.Element =>
{
    const classes = UserSignupStyle();
    return (
        <section className={classes.section}>
            <Container maxWidth="sm">
                <Box pt={18} pb={10}>
                    <Card elevation={0} className={classes.card}>
                        <CardContent className={classes.card_content}>
                            <Box mb={3} textAlign="center">
                            <AccountCircle className={classes.account} />
                                <Typography className={classes.caption}>
                                    {props.bind?.isLoading 
                                    ? <Skeleton variant="text" /> 
                                    : props.bind?.caption}
                                </Typography>
                            </Box>
                            <Box>
                                <Grid container spacing={2}>
                                    <Grid item xs={12} sm={6}>
                                        {props.bind?.isLoading 
                                        ? <Skeleton variant="rect" width="100%" height="45px" /> 
                                        : <TextField 
                                            required 
                                            fullWidth 
                                            id="firstName" 
                                            name="firstName" 
                                            variant="outlined" 
                                            autoComplete="fname" 
                                            onKeyDown={props.bind?.keyHandler}
                                            onChange={props.bind?.formHandler} 
                                            value={props.bind?.firstName} 
                                            label={props.bind?.labelFirstName}
                                        />}
                                    </Grid>
                                    <Grid item xs={12} sm={6}>
                                        {props.bind?.isLoading 
                                        ? <Skeleton variant="rect" width="100%" height="40px" />
                                        : <TextField 
                                            required 
                                            fullWidth 
                                            id="lastName" 
                                            name="lastName" 
                                            variant="outlined" 
                                            autoComplete="lname"
                                            onKeyDown={props.bind?.keyHandler}
                                            onChange={props.bind?.formHandler} 
                                            value={props.bind?.lastName} 
                                            label={props.bind?.labelLastName}
                                        />}
                                    </Grid>
                                    <Grid item xs={12}>
                                        {props.bind?.isLoading 
                                        ? <Skeleton variant="rect" width="100%" height="45px" />
                                        : <TextField 
                                            required 
                                            fullWidth 
                                            id="email" 
                                            name="email" 
                                            variant="outlined" 
                                            autoComplete="email"
                                            onKeyDown={props.bind?.keyHandler}
                                            onChange={props.bind?.formHandler} 
                                            value={props.bind?.email} 
                                            label={props.bind?.labelEmail}
                                        />}
                                    </Grid>
                                    <Grid item xs={12}>
                                        {props.bind?.isLoading 
                                        ? <Skeleton variant="rect" width="100%" height="45px" />
                                        : <TextField 
                                            required 
                                            fullWidth 
                                            id="password" 
                                            name="password" 
                                            variant="outlined" 
                                            type="password" 
                                            autoComplete="current-password"
                                            onKeyDown={props.bind?.keyHandler}
                                            onChange={props.bind?.formHandler} 
                                            value={props.bind?.password} 
                                            label={props.bind?.labelPassword}
                                        />}
                                    </Grid>
                                    <Grid item xs={12}>
                                        {props.bind?.isLoading 
                                        ? <Skeleton variant="rect" width="100%" height="30px" />
                                        : <FormControlLabel 
                                            control={<VioletCheckbox 
                                                id="terms" 
                                                name="terms" 
                                                onChange={props.bind?.formHandler} 
                                                checked={props.bind?.terms} 
                                            />} 
                                            label={props.bind?.consent}
                                        />}
                                    </Grid>
                                </Grid>
                                <Box my={2}>
                                    {props.bind?.isLoading 
                                    ? <Skeleton variant="rect" width="100%" height="40px" /> 
                                    : <ActiveButton {...props} />}
                                </Box>
                                <Box textAlign="right">
                                    {props.bind?.isLoading 
                                    ? <Skeleton variant="text" /> 
                                    : <RedirectTo path="/signin" name={props.bind?.link} />}
                                </Box>
                            </Box>
                        </CardContent>
                    </Card>
                </Box>
            </Container>
        </section>  
    );
}
