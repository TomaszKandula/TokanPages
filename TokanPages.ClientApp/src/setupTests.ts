// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
