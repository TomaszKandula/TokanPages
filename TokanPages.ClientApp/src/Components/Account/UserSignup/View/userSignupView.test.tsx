import "../../../../setupTests";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import { UserSignupView } from "./userSignupView";

describe("test account group component: userSignupView", () => {
    it("should render correctly '<UserSignupView />' when content is loaded.", () => {
        const html = render(
            <BrowserRouter>
                <UserSignupView
                    isLoading={false}
                    caption="Create a new account"
                    warning="Please use..."
                    consent="I agree to the terms of use and privacy policy."
                    button="Sign up"
                    link={{
                        text: "Already have an account? Sign in",
                        href: "/signin"
                    }}
                    buttonHandler={jest.fn()}
                    keyHandler={jest.fn()}
                    formHandler={jest.fn()}
                    progress={false}
                    firstName="Ester"
                    lastName="Exposito"
                    email="ester.exposito@gmail.com"
                    password="madrilena123"
                    terms={false}
                    labelFirstName="First name"
                    labelLastName="Last name"
                    labelEmail="Email address"
                    labelPassword="Password"
                />
            </BrowserRouter>
        );

        expect(html).toMatchSnapshot();
    });
});
