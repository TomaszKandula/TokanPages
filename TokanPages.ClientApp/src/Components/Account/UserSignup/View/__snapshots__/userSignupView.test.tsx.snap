// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test account group component: userSignupView should render correctly '<UserSignupView />' when content is loaded. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="section"
      >
        <div
          class="MuiContainer-root container MuiContainer-maxWidthLg"
        >
          <div
            class="MuiBox-root MuiBox-root-1"
          >
            <div
              class="MuiPaper-root MuiCard-root card MuiPaper-elevation0 MuiPaper-rounded"
            >
              <div
                class="MuiCardContent-root card-content"
              >
                <div
                  class="MuiBox-root MuiBox-root-2"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root account"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                    />
                  </svg>
                  <p
                    class="MuiTypography-root caption MuiTypography-body1"
                  >
                    Create a new account
                  </p>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-3"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
                    >
                      <div
                        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-focused Mui-focused Mui-required Mui-required"
                          data-shrink="true"
                          for="firstName"
                          id="firstName-label"
                        >
                          First name
                          <span
                            aria-hidden="true"
                            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                          >
                             
                            *
                          </span>
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
                        >
                          <input
                            aria-invalid="false"
                            autocomplete="one-time-code"
                            class="MuiInputBase-input MuiOutlinedInput-input"
                            id="firstName"
                            name="firstName"
                            required=""
                            type="text"
                            value="Ester"
                          />
                          <fieldset
                            aria-hidden="true"
                            class="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="PrivateNotchedOutline-legendLabelled-6 PrivateNotchedOutline-legendNotched-7"
                            >
                              <span>
                                First name
                                 *
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
                    >
                      <div
                        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                          data-shrink="true"
                          for="lastName"
                          id="lastName-label"
                        >
                          Last name
                          <span
                            aria-hidden="true"
                            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                          >
                             
                            *
                          </span>
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                        >
                          <input
                            aria-invalid="false"
                            autocomplete="one-time-code"
                            class="MuiInputBase-input MuiOutlinedInput-input"
                            id="lastName"
                            name="lastName"
                            required=""
                            type="text"
                            value="Exposito"
                          />
                          <fieldset
                            aria-hidden="true"
                            class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="PrivateNotchedOutline-legendLabelled-10 PrivateNotchedOutline-legendNotched-11"
                            >
                              <span>
                                Last name
                                 *
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <div
                        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                          data-shrink="true"
                          for="email"
                          id="email-label"
                        >
                          Email address
                          <span
                            aria-hidden="true"
                            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                          >
                             
                            *
                          </span>
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                        >
                          <input
                            aria-invalid="false"
                            autocomplete="one-time-code"
                            class="MuiInputBase-input MuiOutlinedInput-input"
                            id="email"
                            name="email"
                            required=""
                            type="text"
                            value="ester.exposito@gmail.com"
                          />
                          <fieldset
                            aria-hidden="true"
                            class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="PrivateNotchedOutline-legendLabelled-14 PrivateNotchedOutline-legendNotched-15"
                            >
                              <span>
                                Email address
                                 *
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                        style="margin: 0px;"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                          data-shrink="true"
                          for="password"
                        >
                          Password
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                            id="password"
                            name="password"
                            type="password"
                            value="madrilena123"
                          />
                          <div
                            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                          >
                            <button
                              aria-label="toggle password visibility"
                              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                              tabindex="0"
                              type="button"
                            >
                              <span
                                class="MuiIconButton-label"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                                  />
                                </svg>
                              </span>
                              <span
                                class="MuiTouchRipple-root"
                              />
                            </button>
                          </div>
                          <fieldset
                            aria-hidden="true"
                            class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
                            style="padding-left: 8px;"
                          >
                            <legend
                              class="PrivateNotchedOutline-legend-17"
                              style="width: 60.5px;"
                            >
                              <span>
                                ​
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <div
                        class="MuiPaper-root MuiAlert-root MuiAlert-standardWarning MuiPaper-elevation0"
                        role="alert"
                      >
                        <div
                          class="MuiAlert-icon"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
                            />
                          </svg>
                        </div>
                        <div
                          class="MuiAlert-message"
                        >
                          <div>
                            Please use...
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <label
                        class="MuiFormControlLabel-root"
                      >
                        <span
                          aria-disabled="false"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-20 MuiCheckbox-root MuiCheckbox-colorSecondary violet-check-box MuiIconButton-colorSecondary"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <input
                              class="PrivateSwitchBase-input-23"
                              data-indeterminate="false"
                              id="terms"
                              name="terms"
                              type="checkbox"
                              value=""
                            />
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                              />
                            </svg>
                          </span>
                        </span>
                        <span
                          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                        >
                          I agree to the terms of use and privacy policy.
                        </span>
                      </label>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root MuiBox-root-24"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained button MuiButton-fullWidth"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Sign up
                      </span>
                    </button>
                  </div>
                  <div
                    class="MuiBox-root MuiBox-root-25"
                  >
                    <a
                      href="/signin"
                    >
                      Already have an account? Sign in
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="section"
    >
      <div
        class="MuiContainer-root container MuiContainer-maxWidthLg"
      >
        <div
          class="MuiBox-root MuiBox-root-1"
        >
          <div
            class="MuiPaper-root MuiCard-root card MuiPaper-elevation0 MuiPaper-rounded"
          >
            <div
              class="MuiCardContent-root card-content"
            >
              <div
                class="MuiBox-root MuiBox-root-2"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root account"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                  />
                </svg>
                <p
                  class="MuiTypography-root caption MuiTypography-body1"
                >
                  Create a new account
                </p>
              </div>
              <div
                class="MuiBox-root MuiBox-root-3"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-focused Mui-focused Mui-required Mui-required"
                        data-shrink="true"
                        for="firstName"
                        id="firstName-label"
                      >
                        First name
                        <span
                          aria-hidden="true"
                          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                        >
                           
                          *
                        </span>
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
                      >
                        <input
                          aria-invalid="false"
                          autocomplete="one-time-code"
                          class="MuiInputBase-input MuiOutlinedInput-input"
                          id="firstName"
                          name="firstName"
                          required=""
                          type="text"
                          value="Ester"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="PrivateNotchedOutline-legendLabelled-6 PrivateNotchedOutline-legendNotched-7"
                          >
                            <span>
                              First name
                               *
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                        data-shrink="true"
                        for="lastName"
                        id="lastName-label"
                      >
                        Last name
                        <span
                          aria-hidden="true"
                          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                        >
                           
                          *
                        </span>
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                      >
                        <input
                          aria-invalid="false"
                          autocomplete="one-time-code"
                          class="MuiInputBase-input MuiOutlinedInput-input"
                          id="lastName"
                          name="lastName"
                          required=""
                          type="text"
                          value="Exposito"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="PrivateNotchedOutline-legendLabelled-10 PrivateNotchedOutline-legendNotched-11"
                          >
                            <span>
                              Last name
                               *
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                        data-shrink="true"
                        for="email"
                        id="email-label"
                      >
                        Email address
                        <span
                          aria-hidden="true"
                          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                        >
                           
                          *
                        </span>
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                      >
                        <input
                          aria-invalid="false"
                          autocomplete="one-time-code"
                          class="MuiInputBase-input MuiOutlinedInput-input"
                          id="email"
                          name="email"
                          required=""
                          type="text"
                          value="ester.exposito@gmail.com"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="PrivateNotchedOutline-legendLabelled-14 PrivateNotchedOutline-legendNotched-15"
                          >
                            <span>
                              Email address
                               *
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth"
                      style="margin: 0px;"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                        data-shrink="true"
                        for="password"
                      >
                        Password
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                          id="password"
                          name="password"
                          type="password"
                          value="madrilena123"
                        />
                        <div
                          class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                        >
                          <button
                            aria-label="toggle password visibility"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
                          style="padding-left: 8px;"
                        >
                          <legend
                            class="PrivateNotchedOutline-legend-17"
                            style="width: 60.5px;"
                          >
                            <span>
                              ​
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  >
                    <div
                      class="MuiPaper-root MuiAlert-root MuiAlert-standardWarning MuiPaper-elevation0"
                      role="alert"
                    >
                      <div
                        class="MuiAlert-icon"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
                          />
                        </svg>
                      </div>
                      <div
                        class="MuiAlert-message"
                      >
                        <div>
                          Please use...
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  >
                    <label
                      class="MuiFormControlLabel-root"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-20 MuiCheckbox-root MuiCheckbox-colorSecondary violet-check-box MuiIconButton-colorSecondary"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <input
                            class="PrivateSwitchBase-input-23"
                            data-indeterminate="false"
                            id="terms"
                            name="terms"
                            type="checkbox"
                            value=""
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                            />
                          </svg>
                        </span>
                      </span>
                      <span
                        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                      >
                        I agree to the terms of use and privacy policy.
                      </span>
                    </label>
                  </div>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-24"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained button MuiButton-fullWidth"
                    tabindex="0"
                    type="submit"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Sign up
                    </span>
                  </button>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-25"
                >
                  <a
                    href="/signin"
                  >
                    Already have an account? Sign in
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
